Code/Framework/AzToolsFramework/Platform/Linux/AzToolsFramework/API/PythonLoader_Linux.cpp

```diff
@@ -30,12 +32,50 @@ namespace AzToolsFramework::EmbeddedPython
         }
     }
 
-    PythonLoader::~PythonLoader()
+    void PythonLoader::UnloadRequiredModules()
     {
         if (m_embeddedLibPythonHandle)
         {
             dlclose(m_embeddedLibPythonHandle);
         }
     }
-    
+
+    AZ::IO::FixedMaxPath PythonLoader::GetDefault3rdPartyPath(bool createOnDemand)
+    {
+        AZ::IO::FixedMaxPath thirdPartyEnvPathPath;
+        static constexpr const char* thirdPartySubpath = ".o3de/3rdParty";
+        const char* envValue = nullptr;
+
+        // First check if `LY_3RDPARTY_PATH` is explicitly set in the environment
+        if ((envValue=std::getenv("LY_3RDPARTY_PATH")) != nullptr)
+        {
+            // If so, then use the path that is set as the third party path
+            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(envValue);
+        }
+        else
+        {
+            // If not, then attempt to get the home path from `getpwuid` by using the current user's id (getuid())
+            // and reading the `*pw_dir` (Initial working directory) as the home directory.
+            envValue = getpwuid(getuid())->pw_dir;
+            AZ_Assert(envValue!=nullptr, "Unable to calculate home directory");
+            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(envValue) / thirdPartySubpath;
+        }
+
+        AZStd::string thirdPartyPathString = thirdPartyEnvPathPath.String();
+        if ((!AZ::IO::FileIOBase::GetDirectInstance()->IsDirectory(thirdPartyPathString.c_str())) && createOnDemand)
```

Comment:
There should be a check that the `AZ::IO::FileIOBase` instance actually exist.
Though I think a better function to use to avoid needing an instance of the `AZ::IO::FileIOBase` is the SystemFile class static function of [IsDirectory](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/SystemFile.h#L109-L110) and [CreateDir](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/SystemFile.h#L128-L129).